symfony/plugins/orangehrmAdminPlugin/Api/JobCategoryAPI.php

```diff
@@ -20,19 +20,29 @@
 namespace OrangeHRM\Admin\Api;
 
 use DaoException;
+use Exception;
 use JobCategoryService;
 use OrangeHRM\Admin\Api\Model\JobCategoryModel;
+use OrangeHRM\Core\Api\V2\CollectionEndpointInterface;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpointInterface;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointCreateResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointDeleteResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointGetAllResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointGetOneResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointUpdateResult;
 use OrangeHRM\Entity\JobCategory;
-use Orangehrm\Rest\Api\EndPoint;
 use Orangehrm\Rest\Api\Exception\RecordNotFoundException;
-use Orangehrm\Rest\Http\Response;
 
-class JobCategoryAPI extends EndPoint
+class JobCategoryAPI extends Endpoint implements CollectionEndpointInterface, ResourceEndpointInterface
```

Comment:
Initially, we don't need it, but later we may need. Since there are some cases we deal with only resources, but not with collections. Anyway we can handle by throwing not implemented exception

Reply:
Added `CrudEndpoint`